diesel/src/pg/connection/result.rs

```diff
@@ -3,38 +3,39 @@ extern crate libc;
 
 use result::{Error, QueryResult, DatabaseErrorInformation, DatabaseErrorKind};
 use super::row::PgRow;
+use super::raw::RawResult;
 
 use self::pq_sys::*;
 use std::ffi::CStr;
 use std::{str, slice};
 
 pub struct PgResult {
-    internal_result: *mut PGresult,
+    internal_result: RawResult,
 }
 
 impl PgResult {
-    pub fn new(internal_result: *mut PGresult) -> QueryResult<Self> {
-        let result_status = unsafe { PQresultStatus(internal_result) };
+    pub fn new(internal_result: RawResult) -> QueryResult<Self> {
```

Comment:
Huh, another `new` method that returns a result. Not as rare as I thought then. Carry on.